python/graphstorm/config/argument.py

```diff
@@ -747,6 +748,142 @@ def eval_fanout(self):
             # By default use -1 as full neighbor
             return [-1] * self.num_layers
 
+    def _check_distill_feats(self, dt_feats):
+        """ Parse features for distillation.
+        """
+        try:
+            dt_feats_dict = {}
+            for val in dt_feats:
+                dt_feats_dict[val.split(":")[0]] = {"id_field": val.split(":")[1].split("@")[0], \
+                    "textual_field": val.split(":")[1].split("@")[1].split("/")}
+
+        except Exception:
+            assert False, f"{fot_name} distill_feats should either in format" \
+                "ntype1:ntype1_id_field@ntype1_textual_field1/ntype1_textual_field2,ntype2:...," \
+                "for example, a valid input can be 'query:query@query,asin:asin@item_name/brand'"
+
+        return dt_feats_dict
+
+    @property
+    def distill_feats(self):
+        """ distillation features
+        """
+        if hasattr(self, "_distill_feats"):
+            dt_feats = self._distill_feats.split(",")
+            return self._check_distill_feats(dt_feats)
+        return None
+
+    @property
+    def textual_path(self):
+        """ distillation textual data path
+        """
+        if hasattr(self, "_textual_path"):
+            return self._textual_path
+        return None
+
+    @property
+    def concat_field_name(self):
+        """ filed name for distillation
+        """
+        if hasattr(self, "_concat_field_name"):
+            return self._concat_field_name
+        return None
+
+    @property
+    def tsf_name(self):
+        """ HuggingFace Transformer-based model name for distillation
+        """
+        if hasattr(self, "_tsf_name"):
+            return self._tsf_name
+        return None
+
+    @property
+    def pre_trained_name(self):
+        """ Name of pre-trained HuggingFace model weights for distillation
+        """
+        if hasattr(self, "_pre_trained_name"):
+            return self._pre_trained_name
+        return None
+
+    @property
+    def distill_batch_size(self):
+        """ Batch size for distillation.
+        """
+        if hasattr(self, "_distill_batch_size"):
+            return self._distill_batch_size
+        return None
+
+    @property
+    def need_infer(self):
+        """ Whether to run GNN inference for distillation.
+        """
+        if hasattr(self, "_need_infer"):
+            return self._need_infer
+        return None
+
+    @property
+    def save_student_path(self):
+        """ Saved model path for distillation.
+        """
+        if hasattr(self, "_save_student_path"):
+            return self._save_student_path
+        return None
+
+    @property
+    def distill_lr(self):
+        """ Learning rate for distillation.
+        """
+        if hasattr(self, "_distill_lr"):
+            return self._distill_lr
+        return None
+
+    @property
+    def save_student_interval(self):
```

Comment:
i think you can reuse the existing argument for this as well.

Reply:
will do